prototype_source/maskedtensor_sparsity.rst

```diff
@@ -0,0 +1,217 @@
+      ]
+    )
+    >>> sparse_coo_mt.get_data()
+    tensor(indices=tensor([[0, 1],
+                          [2, 2]]),
+          values=tensor([3, 5]),
+          size=(2, 3), nnz=2, layout=torch.sparse_coo)
+
+Sparse CSR tensors
+------------------
+
+Similarly, :class:`MaskedTensor` also supports the
+`CSR (Compressed Sparse Row) <https://pytorch.org/docs/stable/sparse.html#sparse-csr-tensor>`__
+sparse tensor format. Instead of storing the tuples of the indices like sparse COO tensors, sparse CSR tensors
+aim to decrease the memory requirements by storing compressed row indices.
+In particular, a CSR sparse tensor consists of three 1-D tensors:
+
+* ``crow_indices``: array of compressed row indices with size ``(size[0] + 1,)``. This array indicates which row
+  a given entry in values lives in. The last element is the number of specified elements,
+  while crow_indices[i+1] - crow_indices[i] indicates the number of specified elements in row i.
+* ``col_indices``: array of size ``(nnz,)``. Indicates the column indices for each value.
+* ``values``: array of size ``(nnz,)``. Contains the values of the CSR tensor.
+
+Of note, both sparse COO and CSR tensors are in a `beta <https://pytorch.org/docs/stable/index.html>`__ state.
+
+By way of example:
+
+    >>> mt_sparse_csr = mt.to_sparse_csr()
+    >>> mt_sparse_csr
+    MaskedTensor(
+      [
+        [      --,       --, 3],
+        [      --,       --, 5]
+      ]
+    )
+    >>> mt_sparse_csr.get_data()
+    tensor(crow_indices=tensor([0, 1, 2]),
+          col_indices=tensor([2, 2]),
+          values=tensor([3, 5]), size=(2, 3), nnz=2, layout=torch.sparse_csr)
+
+Appendix
+++++++++
+
+.. _sparse-coo-appendix:
+
+Sparse COO construction
+-----------------------
+
+Recall in our original example, we created a :class:`MaskedTensor` and then converted it to a sparse COO MaskedTensor
+with :meth:`MaskedTensor.to_sparse_coo`.
+
+Alternatively, we can also construct a sparse COO MaskedTensor directly by passing in two sparse COO tensors:
+
+    >>> values = torch.tensor([[0, 0, 3], [4, 0, 5]]).to_sparse()
+    >>> mask = torch.tensor([[False, False, True], [False, False, True]]).to_sparse()
+    >>> mt = masked_tensor(values, mask)
+    >>> values
+    tensor(indices=tensor([[0, 1, 1],
+                          [2, 0, 2]]),
+          values=tensor([3, 4, 5]),
+          size=(2, 3), nnz=3, layout=torch.sparse_coo)
+    >>> mask
+    tensor(indices=tensor([[0, 1],
+                          [2, 2]]),
+          values=tensor([True, True]),
+          size=(2, 3), nnz=2, layout=torch.sparse_coo)
+    >>> mt
+    MaskedTensor(
+      [
+        [      --,       --, 3],
+        [      --,       --, 5]
+      ]
+    )
+
+Instead of using :meth:`torch.Tensor.to_sparse`, we can also create the sparse COO tensors directly,
+which brings us to a warning:
+
+.. warning::
+
+  When using a function like :meth:`MaskedTensor.to_sparse_coo`, if the user does not specify the indices 
```

Comment:
The flow here feels confusing as `to_sparse_coo` wasn't used in the above code block, was this warning supposed to say when creating data/mask with `torch.Tensor.to_sparse`? Or is this trying to create a link between the behavior of tensor.to_sparse and the behavior of `MaskedTensor.to_sparse_coo`?

Reply:
Ah - `to_sparse` creates a sparse COO tensor by default. I'll clarify